File: src/server/models/project.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class Project(BaseModel):
    id: str
    name: str
    path: str
    is_self: bool = False
    active: bool = False
    supervisor_agent_id: Optional[str] = None
    hooks_installed: bool = False
    added_at: Optional[str] = None
    git_remote: Optional[str] = None
    language: Optional[str] = None
    description: Optional[str] = None


class ProjectList(BaseModel):
    projects: list[Project]
    total: int


class ProjectCreate(BaseModel):
    path: str
    name: Optional[str] = None
    description: Optional[str] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    language: Optional[str] = None
    git_remote: Optional[str] = None

File: src/frontend/src/types/project.ts
export interface Project {
  id: string;
  name: string;
  path: string;
  is_self: boolean;
  active: boolean;
  supervisor_agent_id: string | null;
  hooks_installed: boolean;
  added_at: string | null;
  git_remote: string | null;
  language: string | null;
  description: string | null;
}

export interface ProjectList {
  projects: Project[];
  total: number;
}

export interface ProjectCreate {
  path: string;
  name?: string;
  description?: string;
}

export interface ProjectAgent {
  key: string;
  registered_at: string;
  type: string;
  agent_id: string;
  display_name: string;
  role: string;
  project_id: string;
}

export interface ProjectAgentsResponse {
  project_id: string;
  agents: ProjectAgent[];
  total: number;
}
